api_fhir/converters/patientConverter.py

```diff
@@ -0,0 +1,256 @@
+            pass  # TODO typeofid isn't provided, this section should contain logic used to create passport field based on typeofid
+        elif imis_insuree.passport is not None:
+            identifier = cls.build_fhir_identifier(imis_insuree.passport, IDENTIFIER_TYPE_SYSTEM,
+                                                   IDENTIFIER_TYPE_PASSPORT_NUMBER_CODE)
+            identifiers.append(identifier.__dict__)
+
+    @classmethod
+    def build_fhir_identifier(cls, value, type_system, type_code):
+        identifier = Identifier()
+        identifier.use = IdentifierUse.USUAL.value
+        type = cls.build_codeable_concept(type_code, type_system)
+        identifier.type = type.__dict__
+        identifier.value = value
+        return identifier
+
+    @classmethod
+    def build_fhir_birth_date(cls, fhir_patient, imis_insuree):
+        fhir_patient.birthDate = imis_insuree.dob.isoformat()
+
+    @classmethod
+    def build_imis_birth_date(cls, imis_insuree, fhir_patient, errors):
+        if not cls.valid_condition(fhir_patient.get('birthDate') is None,
+                                   'Missing patient `birthDate` attribute', errors):
+            dob = None
+            try:
+                dob = core.datetime.datetime.strptime(fhir_patient.get('birthDate'), ISO_DATE_FORMAT)
+            except ValueError:
+                dob = core.datetime.datetime.strptime(fhir_patient.get('birthDate'), ISO_DATETIME_FORMAT)
+            imis_insuree.dob = dob
+
+    @classmethod
+    def build_fhir_gender(cls, fhir_patient, imis_insuree):
+        if hasattr(imis_insuree, "gender") and imis_insuree.gender is not None:
+            code = imis_insuree.gender.code
+            if code == "M":
+                fhir_patient.gender = AdministrativeGender.MALE.value
+            elif code == "F":
+                fhir_patient.gender = AdministrativeGender.FEMALE.value
+            elif code == "O":
+                fhir_patient.gender = AdministrativeGender.OTHER.value
+        else:
+            fhir_patient.gender = AdministrativeGender.UNKNOWN.value
+
+    @classmethod
+    def build_imis_gender(cls, imis_insuree, fhir_patient):
+        if fhir_patient.get("gender") is not None:
+            gender = fhir_patient.get("gender")
+            imis_gender_code = None
+            if gender == AdministrativeGender.MALE.value:
+                imis_gender_code = "M"
+            elif gender == AdministrativeGender.FEMALE.value:
+                imis_gender_code = "F"
+            elif gender == AdministrativeGender.FEMALE.value:
+                imis_gender_code = "O"
+            if imis_gender_code is not None:
+                imis_insuree.gender = Gender.objects.get(pk=imis_gender_code)
+
+    @classmethod
+    def build_fhir_marital_status(cls, fhir_patient, imis_insuree):
+        if imis_insuree.marital is not None:
+            if imis_insuree.marital == "M":
```

Comment:
you should use constant instead of M. Moreover, why you are using constant only for married status ? Similar approach can be done with others

Reply:
I've created enums for the FHIR models (that's why I use it only for `maritalStatus`) I believe the constants for the other IMIS resources should be put in separate modules. But you are right and I will create some temporary enums for them.

Reply:
changed